install/updates/73-service-rbcd.update

```diff
@@ -0,0 +1,5 @@
+dn: $SUFFIX
+add:aci: (targetattr = "objectclass")(targattrfilters="add=objectclass:(objectclass=resourcedelegation)")(version 3.0;acl "RBCD:Kerberos principals can manage resource-based constrained delegation for themselves";allow (write) userdn = "ldap:///self";)
+add:aci: (targetattr = "objectclass")(targattrfilters="add=objectclass:(objectclass=resourcedelegation)")(version 3.0;acl "RBCD:Managing principals can manage resource-based constrained delegation for other principals";allow (write) userattr = "managedby#GROUPDN" or userattr = "managedby#USERDN";)
+add:aci: (targetattr = "objectclass")(targattrfilters="add=objectclass:(objectclass=resourcedelegation)")(version 3.0;acl "RBCD:Delegated permission to manage resource-based constrained delegation for other principals";allow (write) userattr="ipaAllowedToPerform;write_delegation#GROUPDN" or userattr="ipaAllowedToPerform;write_delegation#USERDN" ;)
+
```

Comment:
I'm not sure about the RBCD prefix. Typically this is "permission". I guess it does make it clear what it is used for but it doesn't follow convention (though this could be a new convention). Similarly, self-service permissions generally use the selfservice prefix.

I can't recall the specifics of delegation but only writing the objectclass?

Reply:
> I can't recall the specifics of delegation but only writing the objectclass?

writing to the objectclass is needed for being able to dynamically add `memberprincipal` attribute for delegation.
The rest is already handled the `ipaallowedoperations` / `ipaallowedtoperform` machinery: you can add a `memberprincipal` value if you have been allowed to operate on this entry and in order to add `memberprincipal` attribute you need to have the `resourcedelegation` objectclass.

Prefixes: may be you are right, I should prefix with `permission:` too.

Reply:
We don't always use `permission:` prefix either -- there are plenty recent ACIs that simply have no prefix at all. See, for example, `install/updates/73-subid.update`. It also gives the most complex ACI with target attribute filters...

Reply:
I think that subid is done incorrectly. The update installs the aci into cn=subids,cn=accounts. The subid plugin itself auto-registers permissions. Since those are in cn=permissions,cn=pbac,$SUFFIX they can be found by permission-find, but the ACIs are not there. Presumably a user that wanted to modify it would find it did nothing, but I didn't try that.
I don't have the answer. Your PR will work in that aci's will most likely work, they just aren't visible to users. If that's ok then I'm fine with it.